### PR TITLE
Readonly lists and dictionaries on response models

### LINKED ISSUE #660: Convention test to ensure all response models use IReadOnlyList<T>
See discussion in https://github.com/octokit/octokit.net/pull/658

### PR DESCRIPTION
**Pull request by khellang**
Closes #660 :dancers:

### CODE REVIEW

**Comment by khellang** [comment]
What's up with these whitespace warnings? :confused: 

> warning: CRLF will be replaced by LF in Octokit/Clients/AssigneesClient.cs.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in Octokit/Clients/FollowersClient.cs.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in Octokit/Clients/GistsClient.cs.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in Octokit/Clients/OAuthClient.cs.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in Octokit/Clients/PullRequestsClient.cs.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in Octokit/Clients/RepoCollaboratorsClient.cs.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in Octokit/Clients/StarredClient.cs.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in Octokit/Clients/TeamsClient.cs.
The file will have its original line endings in your working directory.

**Comment by shiftkey** [comment]
@khellang SourceLink requires the files to end in `LF` to index correctly - so if you don't have `git config --global core.autocrlf input` set you're just gonna keep seeing those warnings.

I thought I'd documented that somewhere, but I've added a reminder https://github.com/octokit/octokit.net/issues/756

**Comment by khellang** [comment]
Is that a good setting to use, globally? Isn't there some way to set it and check it into the repo?

**Comment by shiftkey** [comment]
@khellang I've just enabled it for this repository, but I'll elaborate on the setup further in that issue.

**Comment by shiftkey** [comment]
Just one failing test to look at:

```
Octokit.Tests.Clients.StatisticsClientTests+TheGetCommitActivityForTheLastYearMethod.RequestsCorrectUrl [FAIL]
NSubstitute.Exceptions.ReceivedCallsException : Expected to receive a call matching:
    GetQueuedOperation<IEnumerable`1>(/repos/username/repositoryName/stats/commit_activity, any CancellationToken)
    Actually received no matching calls.
```

**Comment by khellang** [comment]
WTF? :confused: 

> Starting Target: SourceLink  
source linking C:\projects\octokit-net\Octokit\bin\Release\Net45\Octokit.pdb 
1 source files do not have matching checksums in the git repository 
no checksum match found for C:\projects\octokit-net\Octokit\Clients\MiscellaneousClient.cs 
Running build failed.
Error:
System.Exception: 1 source files do not have matching checksums in the git repository
   at FSI_0001.SourceLink.GitRepo.VerifyChecksums(GitRepo x, IEnumerable`1 files) in C:\projects\octokit-net\tools\SourceLink.Fake\tools\SourceLink.fsx:line 36
   at FSI_0001.Build.clo@103-16.Invoke(String pf) in C:\projects\octokit-net\build.fsx:line 107
   at Microsoft.FSharp.Collections.SeqModule.Iterate[T](FSharpFunc`2 action, IEnumerable`1 source)
   at FSI_0001.Build.clo@97-15.Invoke(Unit _arg9) in C:\projects\octokit-net\build.fsx:line 99
   at Fake.TargetHelper.runTarget@317(String targetName) in D:\code\fake\src\app\FakeLib\TargetHelper.fs:line 331

**Comment by Haacked** [comment]
@shiftkey is looking into this. At this point, I think we should temporarily stop using SourceLink. @khellang, mind removing the following two lines and pushing again and see if it builds correctly?

https://github.com/octokit/octokit.net/blob/master/build.fsx#L189-L190

**Comment by khellang** [comment]
I give up :sob:

**Comment by shiftkey** [comment]
Don't worry about it @khellang, I tracked this down to https://github.com/octokit/octokit.net/commit/37534f9c0c61d5a317430e63c17ee712595316f2. You can cherry-pick that in...

**Comment by khellang** [comment]
Ok, here we go again... :smile:

**Comment by khellang** [comment]
Phew! :sweat:

File: Octokit/Models/Response/CommitActivity.cs [hunk 1]
```diff
@@ -13,13 +14,15 @@ public class CommitActivity
 
         public CommitActivity(IEnumerable<WeeklyCommitActivity> activity)
         {
-            Activity = activity;
+            Ensure.ArgumentNotNull(activity, "activity");
```
Comment (shiftkey) [inline Octokit/Models/Response/CommitActivity.cs]: :sparkles:

Comment (khellang) [inline Octokit/Models/Response/CommitActivity.cs]: I guess I should've added these to most (all?) parameters in the other response class ctors as well, but wanted to keep to the issue at hand. Do we care?

Comment (Haacked) [inline Octokit/Models/Response/CommitActivity.cs]: I really want to get Fody.NullGuard integrated into this. So I do care, but not that much.

**Pull request merged**